pkg/endpoint/endpoint.go

```diff
@@ -1091,6 +1091,10 @@ func (e *Endpoint) leaveLocked(proxyWaitGroup *completion.WaitGroup, conf Delete
 		e.owner.Datapath().Loader().Unload(e.createEpInfoCache(""))
 	}
 
+	// Remove policy references from shared policy structures
+	e.desiredPolicy.Detach()
+	e.realizedPolicy.Detach()
```

Comment:
Actually, could this remove the same reference twice?

Reply:
I guess not:

```
	if l4.users != nil {
		delete(l4.users, user)
	}
```

Reply:
It could try, but `delete()` allows for it.